ext/pgsql/pgsql.c

```diff
@@ -587,12 +589,25 @@ static void php_pgsql_do_connect(INTERNAL_FUNCTION_PARAMETERS, int persistent)
 			if (zend_register_persistent_resource(ZSTR_VAL(str.s), ZSTR_LEN(str.s), pgsql, le_plink) == NULL) {
 				goto err;
 			}
-			PGG(num_links)++;
-			PGG(num_persistent)++;
+			if (!update) {
+				PGG(num_links)++;
+				PGG(num_persistent)++;
+			}
```

Comment:
Does this "if" mean that it will not increment if the connection is closed in L600 to L610?

Personally, I think it would be better to decrement when the connection is closed in L600 to L610, rather than placing an "if" here.